internal/common/function/function.go

```diff
@@ -314,6 +314,79 @@ func GetBackupQueueInfo(ctx context.Context, clientInstance client.Client, names
 	return queueInfo, nil
 }
 
+// GetActiveVeleroRestoresByLabel retrieves all VeleroRestore objects based on a specified label within a given namespace.
+// It returns a slice of VeleroRestore objects or nil if none are found.
+func GetActiveVeleroRestoresByLabel(ctx context.Context, clientInstance client.Client, namespace, labelKey, labelValue string) ([]velerov1.Restore, error) {
+	var veleroRestoreList velerov1.RestoreList
+	labelSelector := client.MatchingLabels{labelKey: labelValue}
+
+	if err := clientInstance.List(ctx, &veleroRestoreList, client.InNamespace(namespace), labelSelector); err != nil {
```

Comment:
I want all the NAC owned restores that does not have `restore.Status.CompletionTimestamp`, which means they are still subject to be handled. We are making queue number based on the list of the objects that are still subject to some work by Velero, the ones that are completed (has the CompletionTimestamp) are not valid anymore.

Will change the implementation to use the `ListObjectsByLabel`.

Reply:
Ah I remember why I could not use `ListObjectsByLabel` It's because velero `RestoreList`:

```
cannot use veleroRestoreList (variable of type "github.com/vmware-tanzu/velero/pkg/apis/velero/v1".RestoreList) as client.ObjectList value in argument to
ListObjectsByLabel: "github.com/vmware-tanzu/velero/pkg/apis/velero/v1".RestoreList does not implement client.ObjectList (method DeepCopyObject has pointer receiver)
```

So I am unable to change this as you requested.

Reply:
Should this also be changed? I don't want this to be any different? This PR/separate ?

https://github.com/migtools/oadp-non-admin/blob/56afada606fe9e8d6ec96d1a1c0228c6d9d9c6fc/internal/common/function/function.go#L246-L267

Reply:
I am ok doing change in this PR as well

Reply:
I will leave as it is in this PR if you agree - changed the backup function and now the tests are failing showing always queue 0. I don't want to spent time in this PR to debug why. Current implementation works as expected.